api/watcher/watcher.go

```diff
@@ -365,6 +373,9 @@ func (w *remoteRelationWatcher) loop(initialChange params.RemoteRelationChangeEv
 		select {
 		// Send out the initial event or subsequent change.
 		case w.out <- change:
+			if w.logger.IsTraceEnabled() {
+				w.logger.Tracef("sent remote relation change %# v", pretty.Formatter(change))
```

Comment:
I'm guessing the IsTraceEnabled for the pretty formatter? Because the only cost of creating the formatter is a reflec.ValueOf which appears fairly cheap.

Reply:
I'm following the conventions used elsewhere which wrap the pretty trace output. I guess the thinking is that every bit adds up in a loaded system.